src/framegrab/rtsp_discovery.py

```diff
@@ -0,0 +1,184 @@
+import logging
+import time
+import urllib.parse
+from enum import Enum
+from typing import List, Optional, Union
+
+import onvif
+from onvif import ONVIFCamera
+from pydantic import BaseModel
+from wsdiscovery import QName
+from wsdiscovery.discovery import ThreadedWSDiscovery as WSDiscovery
+
+logger = logging.getLogger(__name__)
+
+
+# Default credentials to try when connecting to RTSP cameras, used in discover_camera_ips() when try_default_logins=True
+DEFAULT_CREDENTIALS = [
+    ("admin", "admin"),
+    ("", ""),
+    ("admin", ""),
+    ("root", "camera"),
+    ("root", "root"),
+    ("admin", "12345"),
+    ("admin", "123456"),
+    ("admin", "password"),
+    ("user", "user"),
+    ("root", "pass"),
+]
+
+"""
+Enum for camera discovery modes. Options to try different default credentials stored in DEFAULT_CREDENTIALS. 
+Consists of three options:
+    light: only try first two usernames and passwords ("admin:admin" and no username/password).
+    complete_fast: try the entire DEFAULT_CREDENTIALS without delays in between. 
+    complete_slow: try the entire DEFAULT_CREDENTIALS with a delay of 1 seconds in between.
+    Defaults to None.
+"""
+
+
+class AutoDiscoverModes(str, Enum):
+    light = "light"
+    complete_fast = "complete_fast"
+    complete_slow = "complete_slow"
+
+
+# Model for storing ONVIF Device Information
+class ONVIFDeviceInfo(BaseModel):
+    ip: str
+    port: Optional[int] = 80
+    username: Optional[str] = ""
+    password: Optional[str] = ""
+    xaddr: Optional[str] = ""  # ONVIF service address
+    rtsp_urls: Optional[List[str]] = []  # List of rtsp urls for the camera
+
+
+class RTSPDiscovery:
+    """Simple RTSP camera discovery with ONVIF capabilities"""
+
+    @staticmethod
+    def discover_camera_ips(auto_discover_modes: Union[AutoDiscoverModes, None] = None) -> List[ONVIFDeviceInfo]:
```

Comment:
I still feel this method name is a little confusing given that the function returns `ONVIFDeviceInfo` and not just IP addresses. Maybe the method should be `discover_onvif_devices`